pyuavcan/transport/can/media/pythoncan/_pythoncan.py

```diff
@@ -26,25 +26,64 @@ class PythonCANMedia(_media.Media):
     A media interface adapter for `python-can <https://github.com/hardbyte/python-can>`_.
 
     - Usage example for PCAN-USB channel 1 (bitrate = 500k, mtu = 8, Node-ID = 10)::
-
-        --tr="CAN(can.media.pythoncan.PythonCANMedia('pcan','PCAN_USBBUS1',5000000,8),10)"
-
+        --tr="CAN(can.media.pythoncan.PythonCANMedia('pcan:PCAN_USBBUS1',5000000,8),10)"
+    - Usage example for PCAN-USB channel 1 (nom.bitrate = 500k, data.bitrate = 2M, mtu = 8, Node-ID = 10)::
+        --tr="CAN(can.media.pythoncan.PythonCANMedia('pcan:PCAN_USBBUS1',[5000000,2000000],8),10)"
     - Usage example for Kvaser channel 0 (bitrate = 500k, mtu = 8, Node-ID = 10)::
-
-        --tr="CAN(can.media.pythoncan.PythonCANMedia('kvaser','0',5000000,8),10)"
+        --tr="CAN(can.media.pythoncan.PythonCANMedia('kvaser:0',5000000,8),10)"
     """
 
-    def __init__(self, iface_name: str, channel_name: str, bitrate: int, mtu: int) -> None:
+    VALID_FD_ADAPTER_SET = {'pcan', 'kvaser'}
+
+    def __init__(self, iface_name: str, bitrate: typing.Union[int, typing.Tuple[int, int]], mtu: int) -> None:
         self._iface_name = str(iface_name)
-        self._channel_name = str(channel_name)
+        self._conn_name = self._iface_name.split(':')
+        if len(self._conn_name) != 2:
+            raise RuntimeError('Wrong interface name format')
```

Comment:
Please do not use generic error types because proper error handling is part of the API contract. The user may be looking to catch specific types to handle them in a particular way; this is why we have a hierarchy of exception types (described in PyUAVCAN docs). It is also always better to provide more context in the error message to help with troubleshooting.

```suggestion
            raise pyuavcan.transport.InvalidMediaConfigurationError(
                "Interface name %r does not match the format 'interface:channel'" % self._iface_name
            )
```

Speaking of context. It's a pet peeve of mine -- I don't understand the modern tendency in UI/UX design to strip error messages from useful information. This:

![image](https://user-images.githubusercontent.com/3298404/91589578-9f7c1f00-e962-11ea-98a8-20ca57d58d94.png)

Is a huge step back from this:

![image](https://user-images.githubusercontent.com/3298404/91589641-c0dd0b00-e962-11ea-8201-ba344fad912b.png)